[UX] Added SettingsProvider that renders in Project Settings /  Package Manager / Credentials

hybridherbst opened the pull request:
For now this is just a simple button opening the credentials window. Helps with discoverability.

![image](https://user-images.githubusercontent.com/2693840/97647417-79522880-1a52-11eb-9bb0-ddc1aa6f6b93.png)

jespersmith commented:
Agreed, nice for discoverability.

Would it make sense to hide it in versions below 2020? Or add a button to Package Manager that opens the registry editor in 2019?

hybridherbst commented:
I rebased the remaining changes into this one and force-pushed.

For 2018.4-2019.4 it looks like this:
![20201109-214930](https://user-images.githubusercontent.com/2693840/98594894-8fbf7600-22d5-11eb-8506-db3a3d2a6b6d.gif)

For 2020.1+ it looks like this:
![20201109-215002](https://user-images.githubusercontent.com/2693840/98594904-91893980-22d5-11eb-9cf2-e9e408429129.gif)

So: the "Credentials" entry actually has all functionality in all versions; the "Package Manager" entry in 2020 just becomes an alternate way to edit the scoped registries.

-

One detail to note: Unity is currently changing the PackMan Terms of Service and makes more clear what is/isn't allowed;
things like "a list of OpenUPM packages to select from" are explicitly forbidden now.

To err on the side of caution, I added a similar info as Unity shows on 2020+ to the "Credentials" entry:
![image](https://user-images.githubusercontent.com/2693840/98595448-4ae80f00-22d6-11eb-813d-e7612cda43bd.png)

jespersmith commented:
Thank you very much for these changes. I merged them and created a new release on GitHub.

I removed the "Edit credentials" button from the Packages menu and linked Manage Scoped Registries to the preference dialog, also in 2019.4.
